runtime/src/bank.rs

```diff
@@ -1271,13 +1272,41 @@ impl Bank {
             })
             .collect::<Vec<(Pubkey, u64)>>();
 
-        node_stakes.iter().for_each(|(pubkey, staked)| {
-            let rent_to_be_paid =
-                (((staked * rent_to_be_distributed) as f64) / (total_staked as f64)) as u64;
-            let mut account = self.get_account(pubkey).unwrap_or_default();
-            account.lamports += rent_to_be_paid;
-            self.store_account(pubkey, &account);
-        });
+        // Sort first by stake and then by pubkey for determinism
+        node_stakes.sort_by(
+            |(pubkey1, staked1), (pubkey2, staked2)| match staked2.cmp(staked1) {
```

Comment:
this sort doesn't need to be stable, does it?  you should be safe sorting merely by stake...

Reply:
@rob-solana If I sort by stake only, it could lead to un-determinism, as the order might not be guaranteed for node with same pubkey.  Which could result in rare cases, different distribution of leftover lamports.

Reply:
ah, good point.

Reply:
are you testing for that?

Reply:
@rob-solana not specifically. Let me add a separate test case for this, just for my assurance.

Reply:
@rob-solana Added a scenario in test-case.